### PR TITLE
implement basic database structure

### PR DESCRIPTION
**Pull request by Luca-Guettinger**

### CODE REVIEW

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/repository/user/UserRepository.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package ch.lalumamesh.notenverwaltung.repository.user;
+
+import ch.lalumamesh.notenverwaltung.model.user.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    User findByUsername(String username);
```
Comment (oppikar) [inline backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/repository/user/UserRepository.java:7]: Would be better to either user 

```findByUsernameOrThrow(String username)```

or 

```findByUsernameOrNull(String username)``` (this needs to be implemented first if I remember correctly)

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/ExamValidator.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private final GeneralConfiguration generalConfiguration;
+
+    @Autowired
+    public ExamValidator(ExamConfiguration examConfiguration, GeneralConfiguration generalConfiguration) {
+        this.examConfiguration = examConfiguration;
+        this.generalConfiguration = generalConfiguration;
+    }
+
+    @Override
+    public void validate(Exam object, Errors errors) {
```
Comment (oppikar) [inline backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/ExamValidator.java:23]: unify pattern wtf

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/GroupValidator.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import ch.lalumamesh.notenverwaltung.api.IValidator;
+import ch.lalumamesh.notenverwaltung.model.GraduationGroup;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+
+@Component
+public class GroupValidator implements IValidator<GraduationGroup> {
+    @Override
+    public void validate(GraduationGroup object, Errors errors) {
+        if (object.getParentGrad() == null && object.getParentGroup() == null) {
```
Comment (oppikar) [inline backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/GroupValidator.java]: Naming (getParentGrade() ?)

now one must be null. intended?

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/UserGroupValidator.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+public class UserGroupValidator implements IValidator<UserGraduationGroup> {
+    private final GeneralConfiguration generalConfiguration;
+
+    @Autowired
+    public UserGroupValidator(GeneralConfiguration generalConfiguration) {
+        this.generalConfiguration = generalConfiguration;
+    }
+
+    @Override
+    public void validate(UserGraduationGroup object, Errors errors) {
```
Comment (oppikar) [inline backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/UserGroupValidator.java:20]: unify pattern (see other comment)

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/UserSubjectValidator.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+public class UserSubjectValidator implements IValidator<UserSubject> {
+    private final GeneralConfiguration generalConfiguration;
+
+    @Autowired
+    public UserSubjectValidator(GeneralConfiguration generalConfiguration) {
+        this.generalConfiguration = generalConfiguration;
+    }
+
+    @Override
+    public void validate(UserSubject object, Errors errors) {
```
Comment (oppikar) [inline backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/UserSubjectValidator.java:20]: unify pattern

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/GroupValidator.java (hunk 3 above)
Comment (oppikar) [reply backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/GroupValidator.java]: typo

**Review by oppikar — DISMISSED** [review]
see comments - rest lgtm

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/GroupValidator.java (hunk 3 above)
Comment (Luca-Guettinger) [reply backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/GroupValidator.java]: intended, a group can not have two parents, either a group or a graduation, but this is no longer relevant, considering that groups no longer exist.

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/repository/user/UserRepository.java (hunk 1 above)
Comment (Luca-Guettinger) [reply backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/repository/user/UserRepository.java:7]: thats a koltin thing

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/ExamValidator.java (hunk 2 above)
Comment (Luca-Guettinger) [reply backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/ExamValidator.java:23]: done

File: backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/UserGroupValidator.java (hunk 4 above)
Comment (Luca-Guettinger) [reply backend/notenverwaltung/src/main/java/ch/lalumamesh/notenverwaltung/validator/UserGroupValidator.java:20]: done

**Review by oppikar — APPROVED** [review]

**Pull request merged**